### ISSUE TITLE
Electronic Speed Controller

### ISSUE THREAD

**Issue by anijjar** [uvic-auvic/Electronics#17]
Electronic Speed Controllers are devices used to control the speed of a DC motor using a PWM input. From research, I discovered that the basic ESC Blue Robotics provides uses a Trapezial control algorithm to drive the motor. This is not ideal because the current going into the motor is lagging behind the flux, causing the motor to require more current to reach the same speed. 

To improve this inefficiency, we can use an algorithm called field oriented control. This algorithm changes the system from being time-variant, to time-invariant, allowing a PI controller and other methods of control to be used.

Basically, I need a pcb that takes a CAN message (or USB Com port) containing a speed and drive the thruster accordingly using the FOC algorithm. Something like this:

![image](https://user-images.githubusercontent.com/26446969/228775614-dbbc9318-8459-48cd-9dfa-0c2142fb9c4b.png)
